### DISCUSSION TITLE
Alt Idle animations don't work

### DISCUSSION THREAD

**Discussion by ghost** [KadeArchive/Kade-Engine#512] [Q&A] [answered]
I'm making a small mod where senpai gets angry mid song, and I've gotten his up, down, left, and right animations working, but the idle animation is just normal senpai. Can anyone tell me if they know how to fix it? (I know the animations are per-section, I have them set for the first 5 turns in this video)

https://user-images.githubusercontent.com/61301511/118984984-0e176780-b94c-11eb-8d84-69887562539c.mp4

**Comment by ghost** [comment]
`case 'senpai':
				frames = Paths.getSparrowAtlas('weeb/senpai');
				animation.addByPrefix('idle', 'Senpai Idle', 24, false);
				animation.addByPrefix('singUP', 'SENPAI UP NOTE', 24, false);
				animation.addByPrefix('singLEFT', 'SENPAI LEFT NOTE', 24, false);
				animation.addByPrefix('singRIGHT', 'SENPAI RIGHT NOTE', 24, false);
				animation.addByPrefix('singDOWN', 'SENPAI DOWN NOTE', 24, false);
				animation.addByPrefix('idle-alt', 'Angry Senpai Idle', 24, false);
				animation.addByPrefix('singUP-alt', 'Angry Senpai UP NOTE', 24, false);
				animation.addByPrefix('singLEFT-alt', 'Angry Senpai LEFT NOTE', 24, false);
				animation.addByPrefix('singRIGHT-alt', 'Angry Senpai RIGHT NOTE', 24, false);
				animation.addByPrefix('singDOWN-alt', 'Angry Senpai DOWN NOTE', 24, false);

				addOffset('idle');
				addOffset("singUP", 5, 37);
				addOffset("singRIGHT");
				addOffset("singLEFT", 40);
				addOffset("singDOWN", 14);
				addOffset('idle-alt');
				addOffset("singUP-alt", 5, 37);
				addOffset("singRIGHT-alt");
				addOffset("singLEFT-alt", 40);
				addOffset("singDOWN-alt", 14);

				playAnim('idle');

				setGraphicSize(Std.int(width * 6));
				updateHitbox();

				antialiasing = false;`

**Reply by ghost** [reply]
There's my code if it helps

**Reply by isakube** [reply]
idle animations can't have an alternate animation

**Reply by PlasmaOJ15** [reply]
Idle animations do not have alts. Angry senpai is a separate character

**Comment by puyoxyz** [comment, marked as answer]
reiterating what @prokube and @PlasmaOJ15 said (which they should have said in a separate answer, not as a reply)

idle animations don't do alt animations. they're always the normal one

**Reply by ghost** [reply]
ight thx
